registry/default/akasha-ui/copy-to-clipboard.tsx

```diff
@@ -0,0 +1,56 @@
+"use client";
+
+import { useState } from "react";
+import { Check } from "lucide-react";
+
+import { Button } from "@/registry/default/ui/button";
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from "@/registry/default/ui/tooltip";
+
+interface CopyToClipboardProps {
+  textToCopy: string;
+  tooltipText?: string;
+  buttonText?: string;
+  buttonSuccessText?: string;
+  resetDuration?: number;
+}
+
+export default function CopyToClipboard({
+  textToCopy,
+  tooltipText = "Copy to clipboard",
+  buttonText = "Copy to clipboard",
+  buttonSuccessText = "Copied",
+  resetDuration = 5000,
+}: CopyToClipboardProps) {
+  const [copied, setCopied] = useState(false);
+
+  const copyToClipboard = async () => {
+    await navigator.clipboard.writeText(textToCopy);
+    setCopied(true);
+    setTimeout(() => setCopied(false), resetDuration);
+  };
+
+  return (
+    <TooltipProvider>
+      <Tooltip>
+        <TooltipTrigger asChild>
+          <Button variant="default" size="sm" onClick={copyToClipboard}>
```

Comment:
1. Why use button here ? Why not directly listen to the event on the `TooltipTrigger` ?
2. The trigger can be any component ... you are wrapping such components using `CopyToClipboard` 
3. You have a call to action text shown on the tooltip, a success text when item is copied and the text to copy